packages/errors/src/context.ts

```diff
@@ -38,9 +44,22 @@ export type SolanaErrorContext = DefaultUnspecifiedErrorContextToUndefined<{
     [SOLANA_ERROR__FAILED_TO_DECODE_ACCOUNT]: {
         address: string;
     };
+    [SOLANA_ERROR__INCORRECT_BASE58_ADDRESS_BYTE_LENGTH]: {
+        actualLength: number;
+    };
+    [SOLANA_ERROR__INCORRECT_BASE58_ADDRESS_LENGTH]: {
+        actualLength: number;
+    };
     [SOLANA_ERROR__INVALID_KEYPAIR_BYTES]: {
         byteLength: number;
     };
+    [SOLANA_ERROR__MAX_NUMBER_OF_PDA_SEEDS_EXCEEDED]: {
+        maxSeeds: number;
+    };
+    [SOLANA_ERROR__MAX_PDA_SEED_LENGTH_EXCEEDED]: {
+        index: number;
+        maxSeedLength: number;
```

Comment:
> Also, it makes a value the maximum seed length that probably won't change in our lifetimes, at least not without a code change being necessary in web3.js.

Yeah, but I really didn't want to have to introduce two areas where we'd have to change that value, if we ever needed to update the code.

> On the other hand, inlining the seed into the error is maybe a bad idea (privacy?).

Probably safe to avoid this, yeah.

> I'm struggling to think of what this error context could contain that would actually help me fix the bug. Maybe the actual length of the violating seed is a start?

Index and actual length should do, no? I doubt we need more than that.

Reply:
Btw, the `MAX_SEEDS` error gets thrown by `createProgramAddress` (also the case in the Rust library as well). This can be confusing to the downstream user, since if they provide 16 seeds (`MAX_SEEDS`), they'll actually be feeding `createProgramAddress` with 17 seeds, because the bump is included, and they'll get this error.

With the changes I have locally right now, you'd basically get a message "provided: 17" when you can visibly see you only provided 16. Is a descriptive message good enough to elaborate this fact? Or do we want to work out the "max seeds" concept to actually be 15?